### DISCUSSION TITLE
Config file for patterns?

### DISCUSSION THREAD

**Discussion by CalebCloud** [ko1nksm/getoptions#21] [Ideas]
Is there any support to read the patterns from a configuration file? 

Genuinely an incredible library that has simplified menus in any shell script I make from here on out.

**Comment by ko1nksm** [comment]
No. I don't know exactly what you mean by a configuration file, but I think a shell script file will suffice.

See [parser_definition.sh](https://github.com/ko1nksm/getoptions/blob/master/examples/parser_definition.sh) and the usage example below. It is declarative enough, and is a more flexible shell script than a configuration file.

```sh
#!/bin/sh

set -eu

VERSION=0.1

. ./lib/getoptions.sh
. ./lib/getoptions_help.sh
. ./lib/getoptions_abbr.sh
. ./examples/parser_definition.sh

# Define the parse function for option parsing
eval "$(getoptions parser_definition parse "$0")"

parse "$@"          # Option parsing
eval "set -- $REST" # Exclude options from arguments

echo "FLAG: $FLAG"
echo "VERBOSE: $VERBOSE"
echo "PARAM: $PARAM"
echo "OPTION: $OPTION"
echo "VERSION: $VERSION"
```
